### PR TITLE
landing cuml

### PR DESCRIPTION
**Pull request by dcolinmorgan**
branch off main repo, suffix moved, fixed a few lint issues already

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -384,6 +392,12 @@ See `help(g.featurize)` for more options
       g.umap(kind='nodes', X=['col_1', ..., 'col_n'], y=['label', ..., 'other_targets'], ...)
     ```
 
+* `umap(engine="...")` supports multiple implementations. It defaults to using the GPU-accelerated `engine="cuml"` when a GPU is available, resulting in orders-of-magnitude speedups, and falls back to CPU processing via `engine="umap"`.:
+
```
Comment (silkspace) [inline README.md:396]: should the `engine="umap_learn"` above? I see `UMAPEngineConcrete = Literal["cuml", "umap_learn"]` are the options.

File: graphistry/umap_utils.py [hunk 2]
```diff
@@ -119,22 +162,29 @@ def umap_lazy_init(
...
-                local_connectivity=local_connectivity,
-                repulsion_strength=repulsion_strength,
-                negative_sample_rate=negative_sample_rate,
-            )
+        if engine_resolved == "umap_learn":
+            _, _, umap_engine = lazy_umap_import_has_dependancy()
+        elif engine_resolved == "cuml":
+            _, _, umap_engine = lazy_cuml_import_has_dependancy()
+        else:
+            raise ValueError("No umap engine, ensure 'auto', 'umap', or 'cuml', and the library is installed")
```
Comment (silkspace) [inline graphistry/umap_utils.py:175]: `umap_learn` here instead?

**Review by silkspace — CHANGES_REQUESTED** [review]
Looks good. Just some docs changes umap-> umap_learn

**Comment by lmeyerov** [comment]
cool let's merge? :)

later, we can do a follow-on PR that goes along w/ a blogpost + updates the ipynb

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -118,7 +118,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: [3.8, 3.9, '3.10']
+        python-version: [3.8, 3.9, '3.10.6']
```
Comment (lmeyerov) [inline .github/workflows/ci.yml:121]: i think we can back this out now

Comment (dcolinmorgan) [reply .github/workflows/ci.yml]: ouch-- i rolled back to 3.10 and now see lots of red

Comment (lmeyerov) [reply .github/workflows/ci.yml]: Yeah I saw something with pandas throwing a type error, didn't have time to dig, but guessing a few other upstream changes to fix against ... I think we can land this tho, as they are unrelated 

Maybe run tests with some error tolerance / skip type checks locally just to ensure not related to your PR then we can land

Chatting w Alex, I'd like to show you the release process, so we can schedule 20min to do together

**Review by silkspace — APPROVED** [review]

**Pull request merged**